File: 2026/07-July/16-Thursday/solution.cs
using System;

public class Solution
{
    public long GcdSum(int[] nums)
    {
        int n = nums.Length;
        int[] transformed = new int[n];

        int currentMax = 0;

        for (int i = 0; i < n; i++)
        {
            if (nums[i] > currentMax)
                currentMax = nums[i];

            transformed[i] = GCD(nums[i], currentMax);
        }

        Array.Sort(transformed);

        long answer = 0;

        int left = 0;
        int right = n - 1;

        while (left < right)
        {
            answer += GCD(transformed[left], transformed[right]);
            left++;
            right--;
        }

        return answer;
    }

    private int GCD(int a, int b)
    {
        while (b != 0)
        {
            int temp = a % b;
            a = b;
            b = temp;
        }

        return a;
    }
}

File: 2026/07-July/16-Thursday/solution.cpp
#include <vector>
#include <algorithm>
#include <numeric>

using namespace std;

class Solution {
public:
    long long gcdSum(vector<int>& nums) {
        vector<int> transformed;
        transformed.reserve(nums.size());

        int currentMax = 0;

        for (int value : nums) {
            currentMax = max(currentMax, value);
            transformed.push_back(gcd(value, currentMax));
        }

        sort(transformed.begin(), transformed.end());

        long long answer = 0;

        int left = 0;
        int right = (int)transformed.size() - 1;

        while (left < right) {
            answer += gcd(transformed[left], transformed[right]);
            left++;
            right--;
        }

        return answer;
    }
};
